examples/tests/cloud-push-pull/Earthfile

```diff
@@ -0,0 +1,136 @@
+FROM ../../..+earthly-docker
+WORKDIR /test
+
+RUN apk update && \
+    apk add curl python3 bash
+
+all:
+    BUILD +google-artifact-registry
+    BUILD +google-container-repository
+    BUILD +azure-container-registry
+    BUILD +amazon-elastic-container-registry
+
+google-base:    
+    RUN curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-324.0.0-linux-x86_64.tar.gz && \
+        tar -xvzf google-cloud-sdk-324.0.0-linux-x86_64.tar.gz && \
+        ./google-cloud-sdk/install.sh -q && \
+        ls -la /test/google-cloud-sdk
+    
+    ENV PATH $PATH:/test/google-cloud-sdk/bin
+
+    RUN --secret=GCP_KEY=+secrets/earthly-technologies/gcp/ci-cd-key \
+        echo $GCP_KEY > key.json
+
+    RUN gcloud auth activate-service-account --key-file /test/key.json
+
+google-artifact-registry:
+    FROM +google-base
+    COPY google-artifact-repository.earth ./Earthfile
+
+    RUN gcloud auth configure-docker us-west1-docker.pkg.dev
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- --ci --push +push
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- -P +pull
+
+google-container-repository:
+    FROM +google-base
+    COPY google-container-registry.earth ./Earthfile
+
+    RUN gcloud auth configure-docker
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- --ci --push +push
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- -P +pull
+
+azure-container-registry:
+    COPY azure-container-registry.earth ./Earthfile
+   
+   # Note that we dont have to install a cred helper here, also the cred helper wouldnt help here anyways
+    RUN --secret AZ_USERNAME=+secrets/earthly-technologies/azure/ci-cd-username \
+        --secret AZ_PASSWORD=+secrets/earthly-technologies/azure/ci-cd-password \
+        docker login earthlyintegrationtest.azurecr.io --username $AZ_USERNAME --password $AZ_PASSWORD
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- --ci --push +push
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- -P +pull
+
+amazon-base:
+    ENV GLIBC_VER=2.31-r0
+
+    # install glibc compatibility for alpine
```

Comment:
Ugh I hat that I have to do this, but the base image is Alpine-based. 100% a copy-paste job from someone on Stack Overflow. It works, not gonna question it.